src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -261,14 +261,24 @@ inline void JavaThread::set_done_attaching_via_jni() {
 
 inline bool JavaThread::is_exiting() const {
   // Use load-acquire so that setting of _terminated by
-  // JavaThread::exit() is seen more quickly.
+  // JavaThread::set_terminated() is seen more quickly.
   TerminatedTypes l_terminated = Atomic::load_acquire(&_terminated);
-  return l_terminated == _thread_exiting || check_is_terminated(l_terminated);
+  return l_terminated == _thread_exiting ||
+         l_terminated == _thread_gc_barrier_detached ||
+         check_is_terminated(l_terminated);
+}
+
+inline bool JavaThread::is_gc_barrier_detached() const {
+  // Use load-acquire so that setting of _terminated by
+  // JavaThread::set_terminated() is seen more quickly.
+  TerminatedTypes l_terminated = Atomic::load_acquire(&_terminated);
+  return l_terminated == _thread_gc_barrier_detached ||
+         check_is_terminated(l_terminated);
 }
 
 inline bool JavaThread::is_terminated() const {
   // Use load-acquire so that setting of _terminated by
-  // JavaThread::exit() is seen more quickly.
+  // JavaThread::set_terminated() is seen more quickly.
```

Comment:
This comment should have been updated when the code in
JavaThread::exit() was refactored into JavaThread::set_terminated().
I'm doing it as part of this fix for clarity.

Reply:
See above discussion about this comment.
I'm in favor of dropping this comment.